### ISSUE TITLE
Property has no initializer and is not definitely assigned in the constructor which defined it through function

### ISSUE THREAD

**Issue by moonrailgun** [microsoft/TypeScript#41265]
**TypeScript Version:**  v4.1.0-dev.20201026 

**Search Terms:**

has no initializer and is not definitely assigned in the constructor

**Code**

```ts
class Foo {
       bar: string

       constructor() {
              this.init()
       }

       init() {
              this.bar = 'test'
       }
}
```

**Expected behavior:**

Should not throw error, because i call `init` in `constructor` scope. and `bar` must be have data

**Actual behavior:**

Property 'bar' has no initializer and is not definitely assigned in the constructor

**Playground Link:** 
https://www.typescriptlang.org/play?ts=4.1.0-dev.20201026#code/MYGwhgzhAEBiD29oG8BQ0OegIzAJwC5oIAXPASwDsBzVdLDYeS0vAV2BPjwAoBKFPQbDoJABbkIAOirkS-IVgC+dEbPkC0IkeMlTceaAF5oAchIBTUqcWYVSoA

**Related Issues:** 
#21132

**Comment by MartinJohns** [comment]
Duplicate of #21132.

**Comment by moonrailgun** [comment]
@MartinJohns Thanks for you reply, i will be close it.
